### PR TITLE
checkmate - Ready for review 

### PR DESCRIPTION
**Pull request by AlekstheRealist**
Finally finished checkmate. Anything else to add? Or anything to refactor? Let me know

### CODE REVIEW

File: app/models/game.rb [hunk 1]
```diff
@@ -99,4 +88,89 @@ def next_turn!
...
+      pieces.each do |piece|
+        return true if piece.color != king.color && piece.valid_move?(king.x_cord, king.y_cord)
+      end
+    end
+
+    return false
+  end
+
+  def checkmate?
+    return false if method1? || method2? || method3?
```
Comment (ilyakrasnov) [inline app/models/game.rb]: great job separating the 3 different logical paths... now lets name them 😉  using your comments we can give them names based on what they check

return false if king_can_move_out_of_check? || ... || ...

File: app/models/game.rb [hunk 2]
```diff
@@ -99,4 +88,89 @@ def next_turn!
...
+    pieces = Piece.where(game_id: id)
+    pieces.each do |piece|
+      if piece.color != current_turn && piece.valid_move?(king_in_check.x_cord, king_in_check.y_cord)
+        check_piece = piece
+        return check_piece
+      end
+    end
+  end
+
+  def king_in_check
```
Comment (ilyakrasnov) [inline app/models/game.rb]: is this method still needed?

File: spec/models/game_spec.rb [hunk 3]
```diff
@@ -42,6 +42,23 @@
     end
   end
 
+  describe '#checkmate?' do
+    it 'should return true if either king is in checkmate' do
```
Comment (ilyakrasnov) [inline spec/models/game_spec.rb]: is this test passing?

File: spec/models/game_spec.rb [hunk 4]
```diff
@@ -42,6 +42,23 @@
...
+      g = Game.create(id: 1)
+      Piece.destroy_all(game_id: g.id)
```
Comment (ilyakrasnov) [inline spec/models/game_spec.rb]: you could use g.pieces.destroy_all

File: spec/models/game_spec.rb [hunk 5]
```diff
@@ -42,6 +42,23 @@
...
+      King.create(color: "white", type: "King", x_cord: 7, y_cord: 0)
```
Comment (ilyakrasnov) [inline spec/models/game_spec.rb]: Does this king and all other pieces belong to the same game?

**Comment by AlekstheRealist** [comment]
@ilyakrasnov yeah, there's a lot to be refactored. Trying to get down the logic first though :)

**Pull request closed without merging**